### PR TITLE
fix: Use pathlib.Path on the entire message.location.path

### PR DESCRIPTION
**Pull request by christokur**
Fix the support for `# noqa`

## Description

The `postfilter.filter_messages` does not recognize the paths from lint messages as matches for ignored paths and messages because it only matches part of the path and uses the `str` representation to compare against sets that use `pathlib.Path` keys.

## Related Issue

#557 

## Motivation and Context

`# noqa` support is broken and users lost the ability to suppress messages since `1.8.0`

## How Has This Been Tested?

Use project tests.
Tested against local repo's where the bug is observed using `prospector <= 1.8.4` and `# noqa` works as expected after the patch.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] My change requires a change to the dependencies
- [ ] I have updated the dependencies accordingly
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by Pierre-Sassoulas — CHANGES_REQUESTED** [review]
Thank you ! Would you mind adding a regression tests too, please ?

**Comment by christokur** [comment]
> Thank you ! Would you mind adding a regression tests too, please ?

Done. To be fair these tests should have been written when the pathlib refactoring happened.

File: tests/suppression/test_suppression.py [hunk 1]
```diff
@@ -37,3 +37,12 @@ def test_filter_messages(self):
             pros = Prospector(config)
             pros.execute()
             self.assertEqual(0, pros.summary["message_count"])
+
+    def test_filter_messages_negative(self):
+        workdir = Path(__file__).parent / "testdata/test_filter_messages_negative"
+        with patch("setoptconf.source.commandline.sys.argv", ["prospector"]):
```
Comment (carlio) [inline tests/suppression/test_suppression.py:43]: Checkout the `tests/utils.py` module, `patch_execution` and `patch_cli` might do what you need here and might as well reuse it

**Review by Pierre-Sassoulas — APPROVED** [review]
LGTM, sorry for the delay !

File: tests/suppression/test_suppression.py (hunk 1 above)
Comment (Pierre-Sassoulas) [reply tests/suppression/test_suppression.py:43]: @christokur I'm not sure you took carlio's comment into account, could you look into it please ?

Comment (christokur) [reply tests/suppression/test_suppression.py:43]: I took it into account yes
``` 
with patch("setoptconf.source.commandline.sys.argv", ...
```
and 
```
with patch("sys.argv",  ...
```
have different semantics which is why I did not use patch_cli

Comment (christokur) [reply tests/suppression/test_suppression.py]: Ok refactored things but be aware that pyupgrade also made changes to how typing is used

**Pull request merged**